drivers/adc/ad4080/ad4080.c

```diff
@@ -978,8 +978,10 @@ int ad4080_init(struct ad4080_dev **device,
 	if (ret)
 		goto error_spi;
 
-	if (data != AD4080_CHIP_ID)
+	if (data != AD4080_CHIP_ID) {
+		ret = -EFAULT;
```

Comment:
`EFAULT` means that a function argument points to an invalid memory address. Doesn't really fit here.

Reply:
Oh, I referred to a recently created driver, for handling this case - https://github.com/analogdevicesinc/no-OS/blob/master/drivers/adc/ad4858/ad4858.c#L969
Would you recommend using EBADMSG? Or a different error code?

Reply:
I think `EINVAL` would be ok.

Reply:
Fixed this